Studio: Fix extreme window height for long error messages

bcotrim opened the pull request:
## Related issues

- Fixes STU-905

## Proposed Changes

- Modified error message display in dialog boxes to show only the first line of error messages
- Added import for `simplifyErrorForDisplay` utility function from `src/lib/error-formatting.ts`
- Applied `simplifyErrorForDisplay` to error messages in `showErrorMessageBox` function
- Prevents extremely tall dialog windows that can exceed screen height (e.g., 1600px vertical resolution)

## Testing Instructions

- On a site with the 2025 theme installed
- Change WP version to 5.8.12 (modify the source code to allow that)
- Change WP version to latest
- Verify that the error dialog now shows only the first line of the error message instead of full stack traces
- Confirm the dialog window height is reasonable and the close button is accessible
- Verify that the "Open Studio Logs" button still provides access to full error details

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

bcotrim committed:
Studio: Fix extreme window height for long error messages

epeicher reviewed (approved):
Thanks @bcotrim for fixing this! All the errors now will have the short value now 🙌 . I have tested it, and it works as expected. I can see the first line in the dialog and the whole error in the logs. LGTM! :shipit: 


|Dialog | Error log|
|-------|------|
|<img width="2444" height="1484" alt="Image" src="https://github.com/user-attachments/assets/6e0dad31-3b5f-4485-930d-56329f0f9f7d" />| <img width="2862" height="209" alt="Image" src="https://github.com/user-attachments/assets/4d2a1127-d87e-445f-aae7-d46183fb6686" />|

epeicher reviewed (changes requested):
Sorry @bcotrim, I have been able to reproduce a similar error reported in the task and I still see the long height in the dialog. 

What I have done is:
1. On a site with the 2025 theme installed
2. Change WP version to 5.8.12 (I modified the source code to allow that)
3. Change WP version to latest
4. Error displayed

|Error changing WP version | 
|-------|
|<img width="2024" height="2753" alt="CleanShot 2025-10-24 at 13 23 13@2x" src="https://github.com/user-attachments/assets/43fef361-031f-4af2-83da-1e5232f8564f" />|

This is the stack trace of the error in the logs:
<details>
<summary>
Error trace
</summary>
<pre>
[2025-10-24T11:27:01.130Z][erro][ren1] Error changing WordPress version: Error: PHP.run() failed with exit code 255. 

=== Stdout ===
 #!/usr/bin/env php
<br />
<b>Deprecated</b>:  Return type of WP_Hook::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>498</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>511</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>524</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>537</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>548</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>440</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>454</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>468</b><br />
<br />
<b>Deprecated</b>:  Return type of WP_Hook::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in <b>/wordpress/wp-includes/class-wp-hook.php</b> on line <b>485</b><br />
<br />
<b>Fatal error</b>:  Uncaught Error: Cannot access private property WP_SQLite_DB::$allow_unsafe_unquoted_parameters in /wordpress/wp-includes/wp-db.php:648
Stack trace:
#0 /wordpress/wp-includes/wp-db.php(648): wpdb-&gt;__get('allow_unsafe_un...')
#1 /wordpress/wp-content/mu-plugins/sqlite-database-integration/wp-includes/sqlite/class-wp-sqlite-db.php(386): wpdb-&gt;__get('allow_unsafe_un...')
#2 /wordpress/wp-includes/option.php(171): WP_SQLite_DB-&gt;prepare('SELECT option_v...', 'WPLANG')
#3 /wordpress/wp-includes/l10n.php(63): get_option('WPLANG')
#4 /wordpress/wp-includes/l10n.php(137): get_locale()
#5 /wordpress/wp-includes/l10n.php(828): determine_locale()
#6 /wordpress/wp-settings.php(516): load_default_textdomain()
#7 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1374): require('/wordpress/wp-s...')
#8 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1293): WP_CLI\Runner-&gt;load_wordpress()
#9 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php(28): WP_CLI\Runner-&gt;start()
#10 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/bootstrap.php(84): WP_CLI\Bootstrap\LaunchRunner-&gt;process(Object(WP_CLI\Bootstrap\BootstrapState))
#11 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/wp-cli.php(35): WP_CLI\bootstrap()
#12 phar:///tmp/wp-cli.phar/php/boot-phar.php(20): include('phar:///tmp/wp-...')
#13 /tmp/wp-cli.phar(4): include('phar:///tmp/wp-...')
#14 /tmp/run-cli.php(25): require('/tmp/wp-cli.pha...')
#15 {main}
  thrown in <b>/wordpress/wp-includes/wp-db.php</b> on line <b>648</b><br />
<br />
<b>Fatal error</b>:  Uncaught Error: Cannot access private property WP_SQLite_DB::$allow_unsafe_unquoted_parameters in /wordpress/wp-includes/wp-db.php:648
Stack trace:
#0 /wordpress/wp-includes/wp-db.php(648): wpdb-&gt;__get('allow_unsafe_un...')
#1 /wordpress/wp-content/mu-plugins/sqlite-database-integration/wp-includes/sqlite/class-wp-sqlite-db.php(386): wpdb-&gt;__get('allow_unsafe_un...')
#2 /wordpress/wp-includes/option.php(171): WP_SQLite_DB-&gt;prepare('SELECT option_v...', 'WPLANG')
#3 /wordpress/wp-includes/l10n.php(63): get_option('WPLANG')
#4 /wordpress/wp-includes/l10n.php(137): get_locale()
#5 /wordpress/wp-includes/l10n.php(828): determine_locale()
#6 /wordpress/wp-includes/class-wp-fatal-error-handler.php(46): load_default_textdomain()
#7 [internal function]: WP_Fatal_Error_Handler-&gt;handle()
#8 {main}
  thrown in <b>/wordpress/wp-includes/wp-db.php</b> on line <b>648</b><br />


=== Stderr ===
 PHP Deprecated:  Return type of WP_Hook::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 498
PHP Deprecated:  Return type of WP_Hook::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 511
PHP Deprecated:  Return type of WP_Hook::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 524
PHP Deprecated:  Return type of WP_Hook::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 537
PHP Deprecated:  Return type of WP_Hook::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 548
PHP Deprecated:  Return type of WP_Hook::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 440
PHP Deprecated:  Return type of WP_Hook::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 454
PHP Deprecated:  Return type of WP_Hook::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 468
PHP Deprecated:  Return type of WP_Hook::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wordpress/wp-includes/class-wp-hook.php on line 485
PHP Fatal error:  Uncaught Error: Cannot access private property WP_SQLite_DB::$allow_unsafe_unquoted_parameters in /wordpress/wp-includes/wp-db.php:648
Stack trace:
#0 /wordpress/wp-includes/wp-db.php(648): wpdb->__get('allow_unsafe_un...')
#1 /wordpress/wp-content/mu-plugins/sqlite-database-integration/wp-includes/sqlite/class-wp-sqlite-db.php(386): wpdb->__get('allow_unsafe_un...')
#2 /wordpress/wp-includes/option.php(171): WP_SQLite_DB->prepare('SELECT option_v...', 'WPLANG')
#3 /wordpress/wp-includes/l10n.php(63): get_option('WPLANG')
#4 /wordpress/wp-includes/l10n.php(137): get_locale()
#5 /wordpress/wp-includes/l10n.php(828): determine_locale()
#6 /wordpress/wp-settings.php(516): load_default_textdomain()
#7 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1374): require('/wordpress/wp-s...')
#8 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1293): WP_CLI\Runner->load_wordpress()
#9 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php(28): WP_CLI\Runner->start()
#10 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/bootstrap.php(84): WP_CLI\Bootstrap\LaunchRunner->process(Object(WP_CLI\Bootstrap\BootstrapState))
#11 phar:///tmp/wp-cli.phar/vendor/wp-cli/wp-cli/php/wp-cli.php(35): WP_CLI\bootstrap()
#12 phar:///tmp/wp-cli.phar/php/boot-phar.php(20): include('phar:///tmp/wp-...')
#13 /tmp/wp-cli.phar(4): include('phar:///tmp/wp-...')
#14 /tmp/run-cli.php(25): require('/tmp/wp-cli.pha...')
#15 {main}
  thrown in /wordpress/wp-includes/wp-db.php on line 648
PHP Fatal error:  Uncaught Error: Cannot access private property WP_SQLite_DB::$allow_unsafe_unquoted_parameters in /wordpress/wp-includes/wp-db.php:648
Stack trace:
#0 /wordpress/wp-includes/wp-db.php(648): wpdb->__get('allow_unsafe_un...')
#1 /wordpress/wp-content/mu-plugins/sqlite-database-integration/wp-includes/sqlite/class-wp-sqlite-db.php(386): wpdb->__get('allow_unsafe_un...')
#2 /wordpress/wp-includes/option.php(171): WP_SQLite_DB->prepare('SELECT option_v...', 'WPLANG')
#3 /wordpress/wp-includes/l10n.php(63): get_option('WPLANG')
#4 /wordpress/wp-includes/l10n.php(137): get_locale()
#5 /wordpress/wp-includes/l10n.php(828): determine_locale()
#6 /wordpress/wp-includes/class-wp-fatal-error-handler.php(46): load_default_textdomain()
#7 [internal function]: WP_Fatal_Error_Handler->handle()
#8 {main}
  thrown in /wordpress/wp-includes/wp-db.php on line 648

    at onSiteEdit (http://localhost:5173/src/modules/site-settings/edit-site-details.tsx:110:19)

</pre>
</details>

bcotrim committed:
clear error message on update site

bcotrim commented:
@epeicher thanks for finding that error, nice catch!
I've updated the testing instructions to follow your steps 🙇 
I've addressed the issue, can you take another look, please?

epeicher reviewed (approved):
Thanks @bcotrim for fixing that! I have tested again with the same site, and I can see the error nicely formatted, while the logs contain the full trace. LGTM! :shipit: 

<img width="2822" height="1750" alt="CleanShot 2025-10-24 at 15 27 54@2x" src="https://github.com/user-attachments/assets/da6fd1c4-7041-4876-a575-8a108c0e215e" />
